internal/util/logger/logger.go

```diff
@@ -120,116 +123,126 @@ func newNumaLogger(writer *io.Writer, level *int) NumaLogger {
 	sink := &LogSink{l: &zl}
 	ll := logr.New(sink).WithName(loggerDefaultName)
 
-	return NumaLogger{&ll}
+	return &NumaLogger{&ll, lvl}
 }
 
 // WithLogger returns a copy of parent context in which the
 // value associated with logger key is the supplied logger.
-func WithLogger(ctx context.Context, logger NumaLogger) context.Context {
-	return context.WithValue(ctx, loggerKey{}, logger)
+func WithLogger(ctx context.Context, logger *NumaLogger) context.Context {
+	return context.WithValue(ctx, loggerKey{}, logger.DeepCopy())
```

Comment:
It seemed to not successfully save the logger into the context when it was copying the value for some reason so I modified it to a pointer